katdal/test/test_chunkstore_rados.py

```diff
@@ -26,29 +26,22 @@
 
 
 class TestRadosChunkStore(ChunkStoreTestBase):
-    """Tests connecting to an actual RADOS service, implying a slower setup."""
+    """Test Ceph functionality by connecting to an actual RADOS service."""
 
-    def setup(self):
+    @classmethod
+    def setup_class(cls):
         # Look for default Ceph installation but expect a special test pool
         config = '/etc/ceph/ceph.conf'
         pool = 'test_katdal'
         try:
-            self.store = RadosChunkStore.from_config(config, pool)
+            cls.store = RadosChunkStore.from_config(config, pool)
         except (ImportError, StoreUnavailable):
             raise SkipTest('Rados not installed or cluster misconfigured / down')
 
     def array_name(self, path):
         namespace = 'katdal_test_chunkstore_rados'
         return self.store.join(namespace, path)
 
-
-class TestDudRadosChunkStore(object):
-    """Tests that don't need a RADOS connection, only a 'dud' store."""
-
-    def setup(self):
-        if not rados:
-            raise SkipTest('Rados not installed')
-
     def test_store_unavailable(self):
```

Comment:
Not sure this is testing what it says on the tin. Shouldn't you test with a config file that points at a non-existent rados server?

Reply:
"Store unavailable" is quite a broad concept which includes "no auth" and "bad config file". I'm testing the latter, which is sufficient for me to get my ~100% code coverage... They all go through the same top-level RADOS call, so no need to test each of them separately.

Reply:
Although I guess they could raise different exceptions, so maybe still work a check if simple enough.

Reply:
Github wasn't showing me the whole function, so I'd missed the second part of the test. You can ignore my comment.

Reply:
I'm now checking your suggestion too - it bags me two more lines of coverage and an extra exception too :-)